src/logger/winston.ts

```diff
@@ -0,0 +1,118 @@
+/**
+ * @module logger
+ */
+
+import { createLogger, format, Logger, transports } from 'winston';
+import { defaultLogLevel, LogLevel, ILogger, ILoggerOptions } from './interface';
+import chalk from 'chalk';
+
+export class WinstonLogger implements ILogger {
```

Comment:
We don't wan't to use winston as we need something cross OS compatible (storing location and managing rotation on different OS) so don't implement logger interface just yet